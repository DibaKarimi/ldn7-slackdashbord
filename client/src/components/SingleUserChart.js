import { Line } from "react-chartjs-2";

const SingleUserChart = (props) => {
	const state = {
		data: {
			labels: props.label,
			datasets: [
				{
					label: "Messages",
					backgroundColor: "#009900",
					borderColor: "#1e4d2b",
					borderWidth: 2,
					data: props.messagesDataSet,
				},

				{
					label: "Reactions",
					backgroundColor: "#ff8c00",
					borderColor: "#e86100",
					borderWidth: 2,
					data: props.reactionsDataSet,
				},
				{
					label: "Average messages",
					backgroundColor: "#bdda57",
					borderColor: "#1e4d2b",
					borderWidth: 3,
					borderDash: [6, 8],
					borderDashOffset: [1, 2, 3],
					data: props.averageMessages,
				},

				{
					label: "Average reactions",
					backgroundColor: "#ffa812",
					borderColor: "#e86100",
					borderWidth: 3,
					borderDash: [6, 8],
					borderDashOffset: [1, 2, 3],
					data: props.averageReactions,
				},
			],
		},
	};
	const options = {
		responsive: true,
		legend: {
			display: false,
		},
	};
	return (
		<div className="chart">
			<Line data={state.data} options={options} />
		</div>
	);
};

export default SingleUserChart;
