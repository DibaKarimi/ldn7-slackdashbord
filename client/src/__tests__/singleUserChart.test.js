import React from "react";
import SingleUserChart from "../components/SingleUserChart";

import { shallow } from "enzyme";

describe("SingleUserChart", () => {
	it("should render chart for a single user correctly", () => {
		const component = shallow(<SingleUserChart />);

		expect(component).toMatchSnapshot();
	});
});
