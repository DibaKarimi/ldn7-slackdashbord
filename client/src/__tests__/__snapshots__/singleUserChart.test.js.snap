// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleUserChart should render chart for a single user correctly 1`] = `
<div
  className="chart"
>
  <ForwardRef
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "green",
            "borderColor": "green",
            "borderWidth": 1,
            "data": undefined,
            "hoverBackgroundColor": "lightGreen",
            "hoverBorderColor": "lightGreen",
            "label": "Messages",
          },
          Object {
            "backgroundColor": "orange",
            "borderColor": "orange",
            "borderWidth": 1,
            "data": undefined,
            "hoverBackgroundColor": "rgba(255,99,132,0.4)",
            "hoverBorderColor": "rgba(255,99,132,1)",
            "label": "Reactions",
          },
        ],
        "labels": undefined,
      }
    }
    options={
      Object {
        "legend": Object {
          "display": false,
        },
        "responsive": true,
        "type": "bar",
      }
    }
  />
</div>
`;
